### PR TITLE
Dark Mode independent of OS setting; statusbar improvements and dark buttons

### PR DESCRIPTION
**Pull request by adzm**
Still in progress but functional. A bit tricky with some things being able to toggle while Notepad++ is running, but that might be acceptable since we already state changing the experimental options requires restarting the application.

### CODE REVIEW

**Comment by adzm** [comment]
@donho this is what we were hoping for initially - not requiring someone to enable dark mode for the whole system just to get dark mode for Notepad++! This works well as it is now though does not dynamically switch quite right while Notepad++ is running, but never leaves anything in an unusable state. I'm experimenting with other changes but wanted to get this last-functional version up here for now.

File: PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp [hunk 1]
```diff
@@ -848,8 +849,62 @@ INT_PTR CALLBACK FindReplaceDlg::run_dlgProc(UINT message, WPARAM wParam, LPARAM
 			return TRUE;
 		}
 
+		case WM_CTLCOLOREDIT:
```
Comment (mere-human) [inline PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:852]: Isn't it the same code as in your other PR? 
https://github.com/notepad-plus-plus/notepad-plus-plus/pull/9800/files#diff-62585f441298d642a3dbd871af2800eac3606f9fe148372ab119feed21e050b0

Comment (adzm) [reply PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:852]: Yeah, there are a bunch of conflicts if I rebase on the current master, once #9800 is merged I'll rebase this one and mark it as non-draft.

**Comment by adzm** [comment]
FYI this is pretty much ready to go now; unfortunately getting the menus and scrollbar hack to dynamically change between dark and light at runtime is not feasible at the moment, but with these changes we can enable dark mode for the app without needing it set for the OS. Waiting for #9800 and then I can rebase this and get it completed.

**Comment by adzm** [comment]
Merging this into #9800 for simplicity

**Comment by adzm** [comment]
Sorry about the PR open/close song and dance. Rushing to finish this on my lunch break and deal with merge conflicts. Looks like everything is working well now though! @donho ready for review and merge.

**Comment by adzm** [comment]
image

Note the dark buttons in find dialog and the status bar is now also dark there, as well as drawing the size gripper thingy

**Comment by donho** [comment]
# THIS-IS-SO-GOOD!

**Comment by mere-human** [comment]
Looks good!

By the way, @adzm do you have any plans to make it "less dark"?
I mean, use a background color other than black? Or tune it down a little.
For example, Windows Explorer uses grayish colors (except menu bar which is black):
image
The same colors could be used in Notepad++.

The similar case for Visual Studio.
Even GitHub has this setting:
image

**Comment by adzm** [comment]
Yeah there is no reason we can't have a dimmed versus completely dark version. I believe I kept everything in NppDarkMode namespace so it can easily be swapped out mostly for other colors. There was the idea of having them all be themeable like the rest of the colors but that has its own obstacles. At some point this week and/or weekend I'll try to get the new dark mode icons in and see if we can also add a less intense dark mode option too.

**Comment by adzm** [comment]
I am all about picking a dark theme to use for default. I am wary of removing all the other options though since we are using some undocumented things and hooks, figured it might be useful to have those available just in case. But maybe it is worth just keeping it simple as well.

Should we do that stuff in a separate PR? Back when I did this initially I was modifying the style configurator thing to have a separate dark theme section, which is probably what we'd want (a different theme for dark mode vs non).

**Comment by donho** [comment]
> since we are using some undocumented things and hooks, figured it might be useful to have those available just in case. 

We can always keep the different options in `config.xml`. We just remove 3 options from the UI. 

> Should we do that stuff in a separate PR?

I'll merge this PR this night, So please do it in the next PR.

>  I was modifying the style configurator thing to have a separate dark theme section, which is probably what we'd want (a different theme for dark mode vs non).

Then please define this dark theme in the next PR and copy it as `notepad-plus-plus\PowerEditor\installer\themes\darkThemeDefault.xml`, then commit it. So When user switchs to Dark mode via Notepad++ preferences, or via Windows 10 system, `darkThemeDefault.xml` will be used automatically.

It's the same logic for the toolbar and tabbar icons - see https://github.com/notepad-plus-plus/notepad-plus-plus/issues/9742#issuecomment-826182763

**Comment by donho** [comment]
I second what @mere-human said with the different (easier) UI way:

image

The slide bar can be 1-3 (3 options) or 1-4 (4 options), so users have choice & it won't waste too much room.

**Pull request closed without merging**